### PR TITLE
refactor: wp-search

### PR DESCRIPTION
**Pull request by rohitp-bsf**
### Description
- refactor: wp-search

### Screenshots
- NA

### Types of changes
- Converted Class-based components to functional-based.
- Refactored the edit JS file into render and settings.

### How has this been tested?
- NA

### Checklist:
- [x] My code is tested
- [x] My code follows accessibility standards 
- [x] My code has proper inline documentation 
- [x] I've included any necessary tests 
- [x] I've included developer documentation 
- [ ] I've added proper labels to this pull request

### CODE REVIEW

File: src/blocks/wp-search/block.js [hunk 1]
```diff
@@ -2,27 +2,25 @@
...
-import UAGB_Block_Icons from "@Controls/block-icons"
+import './style.scss';
+import './editor.scss';
+import save from './save';
+import edit from './edit';
+import attributes from './attributes';
+import UAGB_Block_Icons from '@Controls/block-icons';
 
-import { __ } from '@wordpress/i18n';
+const { __ } = wp.i18n;
```
Comment (sandeshjangam) [inline src/blocks/wp-search/block.js:12]: It should be
import { __ } from '@wordpress/i18n';

**Review by sandeshjangam — APPROVED** [review]

File: src/blocks/wp-search/edit.js [hunk 2]
```diff
@@ -1,642 +1,82 @@
...
+			block_id: props.clientId.substr( 0, 8 ),
+		} ); // Pushing Style tag for this block css.
+
+		const $style = document.createElement( 'style' );
+		$style.setAttribute(
+			'id',
+			'uagb-style-wp-search-' + props.clientId.substr( 0, 8 )
+		);
+		document.head.appendChild( $style );
+	}, [ props ] );
```
Comment (RajkiranBagal) [inline src/blocks/wp-search/edit.js:38]: For componentDidMount we need `}, [] );`

File: src/blocks/wp-search/render.js [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+					</div>
+				</form>
+			);
+		}
+
+		return '';
+	};
+
+	return (
+		<>
```
Comment (RajkiranBagal) [inline src/blocks/wp-search/render.js:113]: Remove extra `<>`

File: src/blocks/wp-search/styling.js [hunk 4]
```diff
@@ -2,238 +2,320 @@
...
+			generateCSSUnit( boxShadowBlur, 'px' ) +
+			' ' +
+			generateCSSUnit( boxShadowSpread, 'px' ) +
+			' ' +
+			boxShadowColor +
+			' ' +
+			boxShadowPositionCSS,
+	};
+
+	let icon_color = textColor;
```
Comment (RajkiranBagal) [inline src/blocks/wp-search/styling.js:138]: User camelCase variable names

File: src/blocks/wp-search/styling.js [hunk 5]
```diff
@@ -2,238 +2,320 @@
...
+	] = {
+		'background-color': buttonBgColor,
+	};
+	selectors[
+		'.uagb-layout-input-button .uagb-search-form__container .uagb-search-submit:hover'
+	] = {
+		'background-color': buttonBgHoverColor,
+	};
+
+	mobile_selectors = {
```
Comment (RajkiranBagal) [inline src/blocks/wp-search/styling.js:236]: Use camelCase variable names

File: src/blocks/wp-search/styling.js [hunk 6]
```diff
@@ -2,238 +2,320 @@
...
+				buttonFontSizeType
+			),
+			'line-height': generateCSSUnit(
+				buttonLineHeightMobile,
+				buttonLineHeightType
+			),
+		},
+	};
+
+	tablet_selectors = {
```
Comment (RajkiranBagal) [inline src/blocks/wp-search/styling.js:268]: User camelCase variable names

File: src/blocks/wp-search/styling.js [hunk 7]
```diff
@@ -2,238 +2,320 @@
...
+				buttonFontSizeTablet,
+				buttonFontSizeType
+			),
+			'line-height': generateCSSUnit(
+				buttonLineHeightTablet,
+				buttonLineHeightType
+			),
+		},
+	};
+	let styling_css = '';
```
Comment (RajkiranBagal) [inline src/blocks/wp-search/styling.js:299]: User camelCase variable names

**Review by RajkiranBagal — CHANGES_REQUESTED** [review]

File: src/blocks/wp-search/styling.js (hunk 7 above)
Comment (rohitp-bsf) [reply src/blocks/wp-search/styling.js]: Done!

File: src/blocks/wp-search/styling.js (hunk 6 above)
Comment (rohitp-bsf) [reply src/blocks/wp-search/styling.js]: Done!

File: src/blocks/wp-search/styling.js (hunk 4 above)
Comment (rohitp-bsf) [reply src/blocks/wp-search/styling.js]: Done!

File: src/blocks/wp-search/render.js (hunk 3 above)
Comment (rohitp-bsf) [reply src/blocks/wp-search/render.js]: Done!

File: src/blocks/wp-search/edit.js (hunk 2 above)
Comment (rohitp-bsf) [reply src/blocks/wp-search/edit.js]: Done!

**Review by RajkiranBagal — APPROVED** [review]

**Pull request merged**